Bspline dict2

EmAstro opened the pull request:
Hopefully, final version of from_dict and to_dict methods for bspline.
I added "test_bsplinetodict()" into /PypeIt/pypeit/tests/ to check that everything was working.

jhennawi commented on pypeit/core/pydl.py:
```diff
@@ -331,7 +329,7 @@ def from_dict(cls, bspline_dict):
                    nord=bspline_dict['nord'],
                    npoly=bspline_dict['npoly'],
-                   bkpt=np.array(bspline_dict['bkpt']),
+                   fullbkpt=np.array(bspline_dict['breakpoints']),
```
This line is incorrect. fullbkpt is not a class attribute. It is breakpoint. Please look at the source code where self.breakpoints is set.

jhennawi commented on pypeit/tests/test_pyidl.py:
```diff
@@ -0,0 +1,21 @@
+    print('Returns 0. if the breakpoints are the same before and after')
+    print('reading from the dictionary:')
+    print(np.max(np.array(bspline_todict['breakpoints'])-bspline_fromdict.breakpoints))
+
```
Consider using assert here to determine whether the test passed. See for example other tests. 

Joe

kbwestfall commented:
Correct me if I'm wrong, but I think there was more that @EmAstro and @jhennawi wanted to do on this one.

EmAstro commented:
To avoid confusion, I added from_dict as an option for __init__ and corrected test_pydl in the test folder. Now to instantiate a bspline from a dict is:
init_bspline = bspline(x, from_dict=dictionary)

jhennawi commented:
Hi Ema,

I think x should be set to None though: init_bspline = bspline(None,
from_dict=dictionary). You might add that in the docs.

Joe


On Wed, Aug 15, 2018 at 12:54 PM EmAstro <notifications@github.com> wrote:

> To avoid confusion, I added from_dict as an option for *init* and
> corrected test_pydl in the test folder. Now to instantiate a bspline from a
> dict is:
> init_bspline = bspline(x, from_dict=dictionary)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pypeit/PypeIt/pull/436#issuecomment-413163155>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AIXjoxp5nFxvBxwqnf60bfSTATO6RkAzks5uQ_3fgaJpZM4V1faM>
> .
>


-- 
------------------------------------------------
Joseph F. Hennawi
Associate Professor
Department of Physics
Broida Hall, UC Santa Barbara
Santa Barbara, CA 93106-9530
Phone: 805-893-3503
Mobile: 805-450-8697
E-mail: joe@ <joe@physics.ucsb.edu>physics.ucsb.edu
http://web.physics.ucsb.edu/~joe/
enigma.physics.ucsb.edu
------------------------------------------------
